### PR TITLE
chore: missing routes defined with zod

### PR DESCRIPTION
**Pull request by rimiti**
+ missing routes defined with zod
+ code refactored

### CODE REVIEW

File: shared/routes/emails.routes.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        "200": z
+          .object({})
+          .strict(),
+      },
+    },
+  },
+  put: {},
+  delete: {},
+  patch: {},
+} satisfies IRoutesDef
```
Comment (kevbarns) [inline shared/routes/emails.routes.ts]: as const satisfies

File: shared/routes/emails.routes.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      response: {
+        "200": z
+          .object({})
+          .strict(),
+      },
+    },
+  },
+  put: {},
+  delete: {},
+  patch: {},
```
Comment (kevbarns) [inline shared/routes/emails.routes.ts]: a supprimer

File: shared/routes/emails.routes.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { extensions } from "../helpers/zodHelpers/zodPrimitives"
+import { z } from "../helpers/zodWithOpenApi"
+
+import { IRoutesDef } from "./common.routes"
+
+export const zEmailsRoutes = {
+  get: {},
```
Comment (kevbarns) [inline shared/routes/emails.routes.ts]: a supprimer si inutilisé

File: shared/routes/partners.routes.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    "/api/partners/parcoursup/formations": {
+      response: {
+        "200": z.array(z.object({ parcoursup_id: Pick<IEligibleTrainingsForAppointmentSchema, "parcoursup_id"> })),
+      },
+    },
+  },
+  post: {},
+  put: {},
+  delete: {},
+  patch: {},
```
Comment (kevbarns) [inline shared/routes/partners.routes.ts]: a supprimer

File: shared/routes/partners.routes.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+} satisfies IRoutesDef
```
Comment (kevbarns) [inline shared/routes/partners.routes.ts]: as const satisfies

**Review by kevbarns — CHANGES_REQUESTED** [review]

**Review by antoinebigard — APPROVED** [review]

**Review by kevbarns — APPROVED** [review]
👍

**Pull request merged**